event_model/documents/stream_datum.py

```diff
@@ -1,57 +1,73 @@
-from typing import Any, Dict
+from typing import List
 
 from typing_extensions import Annotated, TypedDict
 
 from .generate.type_wrapper import Field, add_extra_schema
 
+
+class StreamRange(TypedDict):
+    """The parameters required to describe a sequence of incrementing integers"""
+
+    start: Annotated[
```

Comment:
I think I've found a use for `step`, we have an excalibur detector that writes round-robin to 4 HDF files, so would be good to say that indices `{40:80}` in fileA map to seq_nums `{161:321:4}`